app/services/arcgis_api/geocoder.rb

```diff
@@ -201,12 +219,17 @@ def request_token
         f: 'json',
       }
 
-      faraday.post(
-        IdentityConfig.store.arcgis_api_generate_token_url,
-        URI.encode_www_form(body),
-      ) do |req|
-        req.options.context = { service_name: 'arcgis_token' }
-      end.body
+      parse_token(
+        faraday.post(
+          IdentityConfig.store.arcgis_api_generate_token_url, URI.encode_www_form(body)
+        ) do |req|
+          req.options.context = { service_name: 'arcgis_token' }
+        end.body,
+      )
```

Comment:
`parse_token` does not seem like a particularly useful method, since it just returns what it takes in, we could inline it with a `tap` since it seems to only use side effects

```suggestion
     faraday.post(
        IdentityConfig.store.arcgis_api_generate_token_url, URI.encode_www_form(body)
      ) do |req|
        req.options.context = { service_name: 'arcgis_token' }
      end.body.tap do |body|
        handle_api_errors(body)
      end
```

Reply:
Taking both bits of feedback, I'm going to essentially re-raise the error and have the controller properly handle the error instead of gobbling it up at this lower level.